packages/core/src/lib/core/types-for-lists.ts

```diff
@@ -286,6 +286,31 @@ function getListsWithInitialisedFields(
           },
         },
         input: { ...f.input }, // copy
+        ui: {
+          ...f.ui,
+          createView: {
+            ...f.ui?.createView,
+            fieldMode: _isEnabled.create
+              ? f.ui?.createView?.fieldMode ?? f.ui?.createView?.defaultFieldMode
+              : 'hidden',
+          },
+
+          itemView: {
+            ...f.ui?.itemView,
+            fieldMode: _isEnabled.update
+              ? f.ui?.itemView?.fieldMode ?? f.ui?.itemView?.defaultFieldMode
+              : _isEnabled.read
+              ? 'read'
+              : 'hidden', // fallback to read only if not omitted
+          },
+
+          listView: {
+            ...f.ui?.listView,
+            fieldMode: _isEnabled.read
+              ? f.ui?.listView?.fieldMode ?? f.ui?.listView?.defaultFieldMode
+              : 'hidden',
+          },
+        },
```

Comment:
I have moved this behaviour to `types-for-lists.ts`, so we can logically group this near to `isEnabled` - `types-for-lists` is not an appropriate name tbh